File: frontend/lib/models/category_config.dart
class CategoryConfig {
  final String id;
  final String name;
  final int order;
  final bool isVisible;

  CategoryConfig({
    required this.id,
    required this.name,
    required this.order,
    required this.isVisible,
  });

  factory CategoryConfig.fromJson(Map<String, dynamic> json) {
    return CategoryConfig(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      order: json['order'] ?? 0,
      isVisible: json['isVisible'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'order': order,
      'isVisible': isVisible,
    };
  }
}

File: backend/src/models/categoryConfig.js
const mongoose = require('mongoose');

const categorySchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
      trim: true,
      unique: true,
    },
    order: {
      type: Number,
      required: true,
      default: 0,
    },
    isVisible: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model('CategoryConfig', categorySchema);
